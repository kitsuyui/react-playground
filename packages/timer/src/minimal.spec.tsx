import { describe, expect, it, vi } from 'vitest'

import { render } from '@testing-library/react'

import { MinimalTimer } from '.'

describe('MinimalTimer', () => {
  it('renders correctly', () => {
    const props = {
      remaining: 1000,
      running: false,
      incrementTimerValue: vi.fn(),
      toggle: vi.fn(),
      reset: vi.fn(),
      start: vi.fn(),
      stop: vi.fn(),
      setTimerValue: vi.fn(),
    }
    const { asFragment } = render(<MinimalTimer {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
