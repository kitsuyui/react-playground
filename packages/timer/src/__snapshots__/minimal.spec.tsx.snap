// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MinimalTimer > renders correctly 1`] = `
<DocumentFragment>
  <span
    style="width: 6em; font-family: monospace;"
  >
    16:40.000
  </span>
  <button
    name="incrementMinute"
    type="button"
  >
    +分
  </button>
  <button
    name="incrementSecond"
    type="button"
  >
    +秒
  </button>
  <button
    name="reset"
    type="submit"
  >
    Reset
  </button>
  <button
    type="submit"
  >
    Start
  </button>
</DocumentFragment>
`;
